src/api/libcellml/reset.h

```diff
@@ -50,114 +50,160 @@ class LIBCELLML_EXPORT Reset: public OrderedEntity
     /**
      * @brief Get the @c Variable for this @c Reset.
      *
-     * Returns the @c Variable of this @c Reset.
+     * Return the @c Variable of this @c Reset.
      *
      * @return The @c Variable.
      */
     VariablePtr variable() const;
 
     /**
-     * @brief Add a child when to this @c Reset.
+     * @brief Set the test @c Variable for this @c Reset.
      *
-     * Add a copy of the given @p when as a child @c When of this @c Reset.
+     * Set the test @c Variable for this @c Reset.
      *
-     * @param when The @c When to add.
+     * @param variable The test @c Variable to set.
      */
-    void addWhen(const WhenPtr &when);
+    void setTestVariable(const VariablePtr &variable);
 
     /**
-     * @brief Remove the @c When at the given @p index.
+     * @brief Get the test variable for this @c Reset.
      *
-     * Remove the @c When with the given @p index. @p index must
-     * be in the range [0, \#Whens).
+     * Return the @c Variable used in the test condition of this @c Reset.
      *
-     * @param index The index of the @c When to remove.
+     * @return The @c Variable used in the test condition.
+     */
+    VariablePtr testVariable() const;
+
+    /**
+     * @brief Append the argument to the test value string for this reset.
+     *
+     * Append @p math to the existing test value string for this reset.
+     *
+     * @param math The @c std::string to append for this reset.
+     */
+    void appendTestValue(const std::string &math);
+
+    /**
+     * @brief Get the test value string for this reset.
+     *
+     * Return a test value string if one has been appended for this reset. Returns
+     * an empty string if a test value has not been appended (or has been removed).
      *
-     * @return True if the @c When was removed, false otherwise.
+     * @return @c std::string test value for this reset.
      */
-    bool removeWhen(size_t index);
+    std::string testValue() const;
 
     /**
-     * @brief Remove the @c When with the given pointer.
+     * @brief Set the test value string for this reset.
      *
-     * Remove the @c When with the pointer @p when.
+     * Set the test value string for this reset. If @p math is an empty
+     * string, the test value will effectively be removed from the reset.
      *
-     * @overload
+     * @param math The @c std::string to append for this reset.
+     */
+    void setTestValue(const std::string &math);
+
+    /**
+     * @brief Clear the test value math for this reset.
+     *
+     * Clear the test value math for this reset.
      *
-     * @param when The pointer to the @c When to remove.
-     * @return True if the @c When was removed, false otherwise.
+     * @sa setTestValue
      */
-    bool removeWhen(const WhenPtr &when);
+    void clearTestValue();
```

Comment:
There was some discussion a while ago about whether to have a clearSomething() function or whether to just use the setSomething("") idiom ... is this consistent with what was decided then? (how's that for the vaguest question ever ... sorry, I can't remember what the outcome was!)

Reply:
I think we went with we should have `clear*` methods, while `set*("")` would still be an option.

Reply:
We don't have a clear method for the math in the component - maybe add a clearMath() method there too for consistency?